graphql/executor.go

```diff
@@ -170,34 +170,42 @@ func PrepareQuery(typ Type, selectionSet *SelectionSet) error {
 				selection.parsed = true
 			}
 
-			if err := PrepareQuery(field.Type, selection.SelectionSet); err != nil {
+			if field.Batch && field.UseBatchFunc(ctx) {
+				selection.UseBatch = true
+			}
+
+			if err := PrepareQuery(ctx, field.Type, selection.SelectionSet); err != nil {
 				return err
 			}
 		}
 		for _, fragment := range selectionSet.Fragments {
-			if err := PrepareQuery(typ, fragment.SelectionSet); err != nil {
+			if err := PrepareQuery(ctx, typ, fragment.SelectionSet); err != nil {
 				return err
 			}
 		}
 		return nil
 
 	case *List:
-		return PrepareQuery(typ.Type, selectionSet)
+		return PrepareQuery(ctx, typ.Type, selectionSet)
 
 	case *NonNull:
-		return PrepareQuery(typ.Type, selectionSet)
+		return PrepareQuery(ctx, typ.Type, selectionSet)
 
 	default:
 		panic("unknown type kind")
 	}
 }
 
-type panicError struct {
```

Comment:
is there any downside to getting rid of this struct?

Reply:
Not being used & private, so... probably not?